libraries/AP_HAL_ChibiOS/hwdef/common/malloc.c

```diff
@@ -85,6 +85,22 @@ void free(void *ptr)
     }
 }
 
+void *realloc(void *ptr, size_t size) {
+    if (size == 0) {
+       free(ptr);
+       return NULL;
+    }
+    if (ptr == NULL) {
+        return malloc(size);
+    }
+    void *new_mem = malloc(size);
+    if (new_mem != NULL) {
+        memcpy(new_mem, ptr, size);
+    }
+    free(ptr);
```

Comment:
if the new malloc fails (returns NULL) then we still free the original ptr? is that right? Also, shouldn't it be nullptr?

Reply:
Good point, it shouldn't be freed in that case, will address `If realloc() fails the original block is left untouched; it is not freed or moved.`

No nullptr in C.